### PR TITLE
task 3

### PR DESCRIPTION
**Pull request by andreikasmykou**
TASK 3 description:

Backend Service is done.
Two separate handlers were created.
Frontend app was integrated with new API Gateway service

Additional scope:
	- Async/await was used in lambda functions
	- ES6 modules were used for product-service implementation
	- Webpack was configured for product-service
	- SWAGGER documentation was created for product-service
	- Lambda handlers (getProductsList, getProductsById) code was written not in 1 single module (file) and separated in codebase.
	- Main error scenarios were handled by API ("Product not found" error, try-catch blocks are used in lambda handlers).

Link to product-service API - https://94ynxyaj90.execute-api.eu-west-1.amazonaws.com/dev/products
Link to product-service API - https://94ynxyaj90.execute-api.eu-west-1.amazonaws.com/dev/products/{productId}

Link to FE MR - https://github.com/Raerka/nodejs-aws-fe/pull/2

### CODE REVIEW

File: product-service/handler.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export { getProductsList } from './src/handlers';
+export { getProductsById } from './src/handlers';
+
+// export const hello: APIGatewayProxyHandler = async (event, _context) => {
```
Comment (KarneyenkaDzmitry) [inline product-service/handler.ts:4]: These comments could be removed.

File: product-service/src/handlers/getProductsList.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      statusCode: 200,
+      body: products,
+      headers: {
+        'Access-Control-Allow-Origin': '*',
+        'Access-Control-Allow-Credentials': true,
+      },
+    };
+  } catch (err) {
+    return {
+      statusCode: err,
```
Comment (KarneyenkaDzmitry) [inline product-service/src/handlers/getProductsList.ts:19]: statusCode: should be a Number, But in your code, it is an Error with an empty message - [15 line](https://github.com/Raerka/nodejs-aws-be/blob/task3_rsschool/product-service/src/service/productService.ts)

**Review by KarneyenkaDzmitry — COMMENTED** [review]
Have you created any unit tests?

File: product-service/serverless.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    apiGateway: {
+      minimumCompressionSize: 1024,
+    },
+    environment: {
+      AWS_NODEJS_CONNECTION_REUSE_ENABLED: '1',
+    },
+  },
+  functions: {
+    getProductsList: {
+      handler: 'handler.getProductsList',
```
Comment (Shutya) [inline product-service/serverless.ts:29]: Just FYI. It's not necessary to keep handler.ts in root directory, you can specify path to lambda function itself, e.g. `handler: src/handlers/getProductsList.getProductsList`

**Review by Shutya — APPROVED** [review]

File: product-service/serverless.ts (hunk 3 above)
Comment (andreikasmykou) [reply product-service/serverless.ts:29]: Got it.
Thanks

File: product-service/src/handlers/getProductsList.ts (hunk 2 above)
Comment (andreikasmykou) [reply product-service/src/handlers/getProductsList.ts]: Fixed. Thank you

File: product-service/handler.ts (hunk 1 above)
Comment (andreikasmykou) [reply product-service/handler.ts]: Fixed.

**Comment by KarneyenkaDzmitry** [comment]
All criteria have been met.
However, the CloudFront endpoint has not been provided to reviewers.
Grade with 11

**Pull request merged**